File: Definitions/Game_Definitions/classes/FactInvoker.lua
---@meta
---@diagnostic disable

---@class FactInvoker : InteractiveMasterDevice
FactInvoker = {}

---@return FactInvoker
function FactInvoker.new() return end

---@param props table
---@return FactInvoker
function FactInvoker.new(props) return end

---@param ri entEntityRequestComponentsInterface
---@return Bool
function FactInvoker:OnRequestComponents(ri) return end

---@param ri entEntityResolveComponentsInterface
---@return Bool
function FactInvoker:OnTakeControl(ri) return end

---@return FactInvokerController
function FactInvoker:GetController() return end

---@return FactInvokerControllerPS
function FactInvoker:GetDevicePS() return end

---@return Bool
function FactInvoker:ShouldAlwaysUpdateDeviceWidgets() return end

File: DecompiledGameScripts/cyberpunk/devices/FactInvoker/factInvoker.swift

public class FactInvoker extends InteractiveMasterDevice {

  protected cb func OnRequestComponents(ri: EntityRequestComponentsInterface) -> Bool {
    EntityRequestComponentsInterface.RequestComponent(ri, n"terminalui", n"worlduiWidgetComponent", false);
    super.OnRequestComponents(ri);
  }

  protected cb func OnTakeControl(ri: EntityResolveComponentsInterface) -> Bool {
    this.m_uiComponent = EntityResolveComponentsInterface.GetComponent(ri, n"terminalui") as worlduiWidgetComponent;
    super.OnTakeControl(ri);
    this.m_controller = EntityResolveComponentsInterface.GetComponent(ri, n"controller") as FactInvokerController;
  }

  public const func GetDevicePS() -> ref<FactInvokerControllerPS> {
    return this.GetController().GetPS();
  }

  protected const func GetController() -> ref<FactInvokerController> {
    return this.m_controller as FactInvokerController;
  }

  public func ShouldAlwaysUpdateDeviceWidgets() -> Bool {
    return true;
  }
}
